src/overlays/actors/ovl_Obj_Iceblock/z_obj_iceblock.c

```diff
@@ -28,158 +62,1538 @@ const ActorInit Obj_Iceblock_InitVars = {
+            Matrix_InsertZRotation_s(this->dyna.actor.shape.rot.z, MTXMODE_APPLY);
+
+            sp34.x = this->unk_248.x;
+            sp34.y = this->unk_248.y - sp24;
+            sp34.z = this->unk_248.z;
+
+            Matrix_MultiplyVector3fByState(&sp34, &sp28);
+
+            sp34.x = this->dyna.actor.world.pos.x;
+            sp34.y = this->dyna.actor.world.pos.y + sp24;
+            sp34.z = this->dyna.actor.world.pos.z;
+
+            Math_Vec3f_Sum(&sp34, &sp28, &temp_s1->world.pos);
+            Matrix_RotateY(this->unk_254.y + this->dyna.actor.home.rot.y, MTXMODE_APPLY);
+            Matrix_InsertXRotation_s(this->unk_254.x + this->dyna.actor.home.rot.x, MTXMODE_APPLY);
+            Matrix_InsertZRotation_s(this->unk_254.z + this->dyna.actor.home.rot.z, MTXMODE_APPLY);
+            Matrix_CopyCurrentState(&sp40);
+            func_8018219C(&sp40, &temp_s1->shape.rot, MTXMODE_APPLY);
+        } else {
+            Math_Vec3f_Sum(&this->dyna.actor.world.pos, &this->unk_248, &temp_s1->world.pos);
+        }
+    }
+}
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32_DIV1000(speedXZ, 16000, ICHAIN_CONTINUE),       ICHAIN_F32_DIV1000(gravity, -1800, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(minVelocityY, -26000, ICHAIN_CONTINUE), ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),      ICHAIN_F32(uncullZoneScale, 150, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 200, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Iceblock/ObjIceblock_Draw.s")
+void ObjIceblock_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjIceblock* this = THIS;
+    Actor* parent;
+    s32 pad2;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->dyna.actor.shape.rot.x = 0;
+    this->dyna.actor.shape.rot.z = 0;
+    this->dyna.actor.world.rot.x = 0;
+    this->dyna.actor.world.rot.z = 0;
+    this->dyna.actor.home.rot.x = 0;
+    this->dyna.actor.home.rot.z = 0;
+    if (!OBJICEBLOCK_GET_2(&this->dyna.actor)) {
+        this->dyna.actor.shape.rot.y = (this->dyna.actor.shape.rot.y + 0x2000) & 0xC000;
+        this->dyna.actor.home.rot.y = this->dyna.actor.shape.rot.y;
+        this->dyna.actor.world.rot.y = this->dyna.actor.shape.rot.y;
+    }
+
+    DynaPolyActor_Init(&this->dyna, 1);
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &D_06000438);
+    func_800C62BC(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+
+    if (OBJICEBLOCK_GET_1(&this->dyna.actor)) {
+        this->collider.dim.yShift = -100;
+        this->collider.dim.height = 126;
+        this->collider.dim.radius = 29;
+    }
+
+    if (D_80A26E7C == NULL) {
+        D_80A26E7C = (AnimatedMaterial*)Lib_SegmentedToVirtual(&D_06000328);
```

Comment:
I think there's no need for this cast
```suggestion
        D_80A26E7C = Lib_SegmentedToVirtual(&D_06000328);
```